src/llama_stack_client/lib/agents/client_tool.py

```diff
@@ -59,8 +60,101 @@ def get_tool_definition(self) -> ToolDefParam:
             tool_prompt_format="python_list",
         )
 
-    @abstractmethod
     def run(
-        self, messages: List[Union[UserMessage, ToolResponseMessage]]
-    ) -> List[Union[UserMessage, ToolResponseMessage]]:
+        self,
+        message: CompletionMessage,
```

Comment:
I actually think it is useful to keep this signature more general to `List[Message]` so a tool can understand more context naturally and be able to act on it.

Reply:
hmm, then we're back to SingleMessageClientTool + ClientTool? https://github.com/meta-llama/llama-stack-client-python/blob/b183fb644cea23068677b55f92fc2f55909fa72e/src/llama_stack_client/lib/agents/agent.py#L81 -- the full message is not being passed currently too

Reply:
no my point is let's just not have any single message thingy - it is not useful. but each run_impl can get a list of messages and pop off the last message. the name of the parameter would be `history` and the sample implementations will look like `assistant_message = history.pop()` and that's it. 

the whole single-message thing is the problem not the signature. I do feel like the return type as a single message is quite sufficient, and does not need to be List

Reply:
got it, updated and added comments for usage around ClientTools. By default we use the last message, but users could override `run` for using entire message history.